testng-eclipse-plugin/src/main/org/testng/eclipse/util/LaunchUtil.java

```diff
@@ -280,7 +280,8 @@ public static void launchMethodConfiguration(IJavaProject javaProject,
         DependencyInfo groupInfo = DependencyInfo.createDependencyInfo(javaProject);
         Set<IMethod> transitiveMethods = findMethodTransitiveClosure(iMethod, groupInfo);
         for (IMethod transitiveMethod : transitiveMethods) {
-          typeAndMethods.add(new TypeAndMethod(iType, transitiveMethod));
+          IType t = transitiveMethod.getDeclaringType();
+          typeAndMethods.add(new TypeAndMethod(t, transitiveMethod));

```

Comment:
use the method's own enclosing type, otherwise, in case of depends on test method in other package, the type would be wrong